### PR TITLE
Add projects

### PR DESCRIPTION
**Pull request by parruda**
- New command `new project :name`
- Refactor project configuration
- On `mod`: move requires inside the `run!` method so it will only happens if the command gets called. That's important to reduce the load time of the CLI.

### CODE REVIEW

File: templates/project/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Gotta Project
+
+This is a Gotta Project!
```
Comment (tylerklose) [inline templates/project/README.md]: Makes me think we should create a badge or something for READMEs

File: lib/gotta/serve.rb [hunk 2]
```diff
@@ -20,14 +20,13 @@
 
 module Gotta
   module Serve
-    require 'gotta/common/project'
+    require 'gotta/project/config'
     # require 'gotta/common/event'
     require 'gotta/serve/response'
     require 'gotta/queue/socket'
 
     # TODO: Fix this weirdness
```
Comment (tylerklose) [inline lib/gotta/serve.rb]: What weirdness?

Comment (tylerklose) [inline lib/gotta/serve.rb]: What weirdness?

File: lib/gotta/run/ruby/loader/component.rb [hunk 3]
```diff
@@ -14,7 +14,7 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
+require 'pry'
```
Comment (tylerklose) [inline lib/gotta/run/ruby/loader/component.rb]: I assume this was left behind?

File: lib/gotta/mod/cli/update.rb [hunk 4]
```diff
@@ -19,25 +19,23 @@ module Gotta
...
-        require 'gotta/project'
-        require 'gotta/mod/repository'
-        require 'open-uri'
-
         command 'update mod :mod_name'
 
-        PROJECT_CONFIG = Gotta::Project.new(Dir.pwd)
-
         def run!
+          require 'gotta/project'
```
Comment (tylerklose) [inline lib/gotta/mod/cli/update.rb]: Should this just be `require 'gotta/project/config'?

File: lib/gotta/mod/cli/uninstall.rb [hunk 5]
```diff
@@ -19,11 +19,10 @@ module Gotta
   module Mod
     module CLI
       class UninstallMod < Gotta::CLI::Command::Base
-        require 'gotta/project'
-
         command 'uninstall mod :mod_name'
 
         def run!
+          require 'gotta/project'
```
Comment (tylerklose) [inline lib/gotta/mod/cli/uninstall.rb]: Should this just be `require 'gotta/project/config'`?

File: lib/gotta/mod/cli/install.rb [hunk 6]
```diff
@@ -19,25 +19,23 @@ module Gotta
...
-        require 'gotta/mod/repository'
-        require 'open-uri'
-
         command 'install mod :mod_name'
 
-        PROJECT_CONFIG = Gotta::Project.new(Dir.pwd)
-
         def run!
+          require 'open-uri'
+          require 'gotta/project/config'
```
Comment (tylerklose) [inline lib/gotta/mod/cli/install.rb]: How come the other `require`s are `require 'gotta/project'` and this one is `require 'gotta/project/config'`?

**Comment by tylerklose** [comment]
Refactoring looks good. I haven't had a chance to pull down the new repo, set up the environment, and test these changes just yet. I left a few comments/questions for ya

File: templates/project/README.md (hunk 1 above)
Comment (parruda) [reply templates/project/README.md]: Yeah. I will create an issue for that!

File: lib/gotta/serve.rb (hunk 2 above)
Comment (parruda) [reply lib/gotta/serve.rb]: I fixed and forgot to remove the comment! 😄

File: lib/gotta/run/ruby/loader/component.rb (hunk 3 above)
Comment (parruda) [reply lib/gotta/run/ruby/loader/component.rb]: Good catch!

File: lib/gotta/mod/cli/update.rb (hunk 4 above)
Comment (parruda) [reply lib/gotta/mod/cli/update.rb]: Good catch, fixed.

File: lib/gotta/mod/cli/install.rb (hunk 6 above)
Comment (parruda) [reply lib/gotta/mod/cli/install.rb]: `'gotta/project/config'` is the correct one.

File: lib/gotta/mod/cli/uninstall.rb (hunk 5 above)
Comment (tylerklose) [reply lib/gotta/mod/cli/uninstall.rb]: @parruda I think one of 
```suggestion
          require 'gotta/project/config'
```

**Comment by tylerklose** [comment]
One last tweak is needed and then this should be good to go

**Comment by parruda** [comment]
@tylerklose I think I got all of them now!

**Comment by tylerklose** [comment]
I ran into one issue:
```console
gotta watch
Traceback (most recent call last):
	12: from /Users/tk/.rvm/gems/ruby-2.6.1/bin/ruby_executable_hooks:24:in `<main>'
	11: from /Users/tk/.rvm/gems/ruby-2.6.1/bin/ruby_executable_hooks:24:in `eval'
	10: from /Users/tk/.rvm/gems/ruby-2.6.1/bin/gotta:23:in `<main>'
	 9: from /Users/tk/.rvm/gems/ruby-2.6.1/bin/gotta:23:in `load'
	 8: from /Users/tk/.rvm/gems/ruby-2.6.1/gems/gotta-0.1.2/exe/gotta:19:in `<top (required)>'
	 7: from /Users/tk/code/source/gotta-dev/gotta-cli/lib/gotta/cli.rb:28:in `run'
	 6: from /Users/tk/code/source/gotta-dev/gotta-cli/lib/gotta/cli.rb:28:in `new'
	 5: from /Users/tk/code/source/gotta-dev/gotta-cli/lib/gotta/cli/command.rb:77:in `initialize'
	 4: from /Users/tk/code/source/gotta-dev/gotta-cli/lib/gotta/cli/command.rb:85:in `build_command'
	 3: from /Users/tk/code/source/gotta-dev/gotta-cli/lib/gotta/cli/command.rb:85:in `each'
	 2: from /Users/tk/code/source/gotta-dev/gotta-cli/lib/gotta/cli/command.rb:94:in `block in build_command'
	 1: from /Users/tk/code/source/gotta-dev/gotta/lib/gotta/mod/cli/watch.rb:26:in `run!'
/Users/tk/code/source/gotta-dev/gotta/lib/gotta/mod.rb:44:in `now': undefined method `folder' for nil:NilClass (NoMethodError)
```

This was solved by making the following change in `project.yml`:
```yaml
project:
  # The project name
  name: email-scheduler
  mods_dir: mods
```
to
```yaml
project:
  # The project name
  name: email-scheduler
  mods:
    folder: mods
```

**Comment by tylerklose** [comment]
I'm getting an error when trying to run `gotta new project new-project`:
```console
gotta new project new-project
Initializing project in 'new-project'
Traceback (most recent call last):
	13: from /Users/tk/.rvm/gems/ruby-2.6.1/bin/ruby_executable_hooks:24:in `<main>'
	12: from /Users/tk/.rvm/gems/ruby-2.6.1/bin/ruby_executable_hooks:24:in `eval'
	11: from /Users/tk/.rvm/gems/ruby-2.6.1/bin/gotta:23:in `<main>'
	10: from /Users/tk/.rvm/gems/ruby-2.6.1/bin/gotta:23:in `load'
	 9: from /Users/tk/.rvm/gems/ruby-2.6.1/gems/gotta-0.1.2/exe/gotta:19:in `<top (required)>'
	 8: from /Users/tk/code/source/gotta-dev/gotta-cli/lib/gotta/cli.rb:28:in `run'
	 7: from /Users/tk/code/source/gotta-dev/gotta-cli/lib/gotta/cli.rb:28:in `new'
	 6: from /Users/tk/code/source/gotta-dev/gotta-cli/lib/gotta/cli/command.rb:77:in `initialize'
	 5: from /Users/tk/code/source/gotta-dev/gotta-cli/lib/gotta/cli/command.rb:85:in `build_command'
	 4: from /Users/tk/code/source/gotta-dev/gotta-cli/lib/gotta/cli/command.rb:85:in `each'
	 3: from /Users/tk/code/source/gotta-dev/gotta-cli/lib/gotta/cli/command.rb:94:in `block in build_command'
	 2: from /Users/tk/code/source/gotta-dev/gotta/lib/gotta/project/cli/new.rb:35:in `run!'
	 1: from /Users/tk/code/source/gotta-dev/gotta/lib/gotta/project/builder.rb:33:in `create'
/Users/tk/code/source/gotta-dev/gotta/lib/gotta/project/builder.rb:46:in `copy_template': uninitialized constant Gotta::GEM_ROOT (NameError)
```

**Comment by parruda** [comment]
> I ran into one issue:
> 
> ```
> gotta watch
> Traceback (most recent call last):
> 	12: from /Users/tk/.rvm/gems/ruby-2.6.1/bin/ruby_executable_hooks:24:in `<main>'
> 	11: from /Users/tk/.rvm/gems/ruby-2.6.1/bin/ruby_executable_hooks:24:in `eval'
> 	10: from /Users/tk/.rvm/gems/ruby-2.6.1/bin/gotta:23:in `<main>'
> 	 9: from /Users/tk/.rvm/gems/ruby-2.6.1/bin/gotta:23:in `load'
> 	 8: from /Users/tk/.rvm/gems/ruby-2.6.1/gems/gotta-0.1.2/exe/gotta:19:in `<top (required)>'
> 	 7: from /Users/tk/code/source/gotta-dev/gotta-cli/lib/gotta/cli.rb:28:in `run'
> 	 6: from /Users/tk/code/source/gotta-dev/gotta-cli/lib/gotta/cli.rb:28:in `new'
> 	 5: from /Users/tk/code/source/gotta-dev/gotta-cli/lib/gotta/cli/command.rb:77:in `initialize'
> 	 4: from /Users/tk/code/source/gotta-dev/gotta-cli/lib/gotta/cli/command.rb:85:in `build_command'
> 	 3: from /Users/tk/code/source/gotta-dev/gotta-cli/lib/gotta/cli/command.rb:85:in `each'
> 	 2: from /Users/tk/code/source/gotta-dev/gotta-cli/lib/gotta/cli/command.rb:94:in `block in build_command'
> 	 1: from /Users/tk/code/source/gotta-dev/gotta/lib/gotta/mod/cli/watch.rb:26:in `run!'
> /Users/tk/code/source/gotta-dev/gotta/lib/gotta/mod.rb:44:in `now': undefined method `folder' for nil:NilClass (NoMethodError)
> ```
> 
> This was solved by making the following change in `project.yml`:
> 
> ```yaml
> project:
>   # The project name
>   name: email-scheduler
>   mods_dir: mods
> ```
> 
> to
> 
> ```yaml
> project:
>   # The project name
>   name: email-scheduler
>   mods:
>     folder: mods
> ```

I've added an example `project.yml` here: https://github.com/gotta-dev/gotta/pull/2/files#diff-f56701aec48fb719da41864644809141

**Comment by parruda** [comment]
> I'm getting an error when trying to run `gotta new project new-project`:
> 
> ```
> gotta new project new-project
> Initializing project in 'new-project'
> Traceback (most recent call last):
> 	13: from /Users/tk/.rvm/gems/ruby-2.6.1/bin/ruby_executable_hooks:24:in `<main>'
> 	12: from /Users/tk/.rvm/gems/ruby-2.6.1/bin/ruby_executable_hooks:24:in `eval'
> 	11: from /Users/tk/.rvm/gems/ruby-2.6.1/bin/gotta:23:in `<main>'
> 	10: from /Users/tk/.rvm/gems/ruby-2.6.1/bin/gotta:23:in `load'
> 	 9: from /Users/tk/.rvm/gems/ruby-2.6.1/gems/gotta-0.1.2/exe/gotta:19:in `<top (required)>'
> 	 8: from /Users/tk/code/source/gotta-dev/gotta-cli/lib/gotta/cli.rb:28:in `run'
> 	 7: from /Users/tk/code/source/gotta-dev/gotta-cli/lib/gotta/cli.rb:28:in `new'
> 	 6: from /Users/tk/code/source/gotta-dev/gotta-cli/lib/gotta/cli/command.rb:77:in `initialize'
> 	 5: from /Users/tk/code/source/gotta-dev/gotta-cli/lib/gotta/cli/command.rb:85:in `build_command'
> 	 4: from /Users/tk/code/source/gotta-dev/gotta-cli/lib/gotta/cli/command.rb:85:in `each'
> 	 3: from /Users/tk/code/source/gotta-dev/gotta-cli/lib/gotta/cli/command.rb:94:in `block in build_command'
> 	 2: from /Users/tk/code/source/gotta-dev/gotta/lib/gotta/project/cli/new.rb:35:in `run!'
> 	 1: from /Users/tk/code/source/gotta-dev/gotta/lib/gotta/project/builder.rb:33:in `create'
> /Users/tk/code/source/gotta-dev/gotta/lib/gotta/project/builder.rb:46:in `copy_template': uninitialized constant Gotta::GEM_ROOT (NameError)
> ```

Fixed here: https://github.com/gotta-dev/gotta/pull/1/files#diff-ff1b63d68e9175862045cba44a88c71dR4

**Comment by tylerklose** [comment]
@parruda looks good to me... it works! The only thing I'm questioning now is the logging that comes after `gotta start`:
```console
gotta new project new-project
Initializing project in 'new-project'
Traceback (most recent call last):
	13: from <redacted>.rvm/gems/ruby-2.6.1/bin/ruby_executable_hooks:24:in `<main>'
	12: from <redacted>/.rvm/gems/ruby-2.6.1/bin/ruby_executable_hooks:24:in `eval'
	11: from <redacted>/.rvm/gems/ruby-2.6.1/bin/gotta:23:in `<main>'
	10: from <redacted>/.rvm/gems/ruby-2.6.1/bin/gotta:23:in `load'
	 9: from <redacted>/.rvm/gems/ruby-2.6.1/gems/gotta-0.1.2/exe/gotta:19:in `<top (required)>'
	 8: from<redacted>/gotta-dev/gotta-cli/lib/gotta/cli.rb:28:in `run'
	 7: from <redacted>/gotta-dev/gotta-cli/lib/gotta/cli.rb:28:in `new'
	 6: from <redacted>/gotta-dev/gotta-cli/lib/gotta/cli/command.rb:77:in `initialize'
	 5: from <redacted>/gotta-dev/gotta-cli/lib/gotta/cli/command.rb:85:in `build_command'
	 4: from <redacted>/gotta-dev/gotta-cli/lib/gotta/cli/command.rb:85:in `each'
	 3: from <redacted>/gotta-dev/gotta-cli/lib/gotta/cli/command.rb:94:in `block in build_command'
	 2: from <redacted>/gotta-dev/gotta/lib/gotta/project/cli/new.rb:35:in `run!'
	 1: from <redacted>/gotta-dev/gotta/lib/gotta/project/builder.rb:33:in `create'
<redacted>/gotta-dev/gotta/lib/gotta/project/builder.rb:46:in `copy_template': uninitialized constant Gotta::GEM_ROOT (NameError)
```

**Comment by parruda** [comment]
@tylerklose this is solved by another PR! I will merge this one

**Pull request merged**